src/v/kafka/server/group_manager.cc

```diff
@@ -127,29 +136,54 @@ ss::future<> group_manager::start() {
     /*
      * periodically remove expired group offsets.
      */
-    _timer.set_callback([this] {
+    _expired_group_offset_timer.set_callback([this] {
         ssx::spawn_with_gate(_gate, [this] {
             return handle_offset_expiration().finally([this] {
                 if (!_gate.is_closed()) {
-                    _timer.arm(_offset_retention_check());
+                    _expired_group_offset_timer.arm(_offset_retention_check());
                 }
             });
         });
     });
-    _timer.arm(_offset_retention_check());
+    _expired_group_offset_timer.arm(_offset_retention_check());
 
     /*
      * reschedule periodic collection of expired offsets when the configured
      * frequency changes. useful if it were accidentally configured to be much
      * longer than desired / reasonable, and then fixed (e.g. 1 year vs 1 day).
      */
     _offset_retention_check.watch([this] {
-        if (_timer.armed()) {
-            _timer.cancel();
-            _timer.arm(_offset_retention_check());
+        if (_expired_group_offset_timer.armed()) {
+            _expired_group_offset_timer.cancel();
+            _expired_group_offset_timer.arm(_offset_retention_check());
         }
     });
 
+    {
+        constexpr auto set_lag_timer = [](group_manager* me) {
+            auto has_lag_metric = enabled_metrics::from_vector(
+                                    me->_enabled_metrics())
+                                    .consumer_lag;
+            me->_lag_metrics_timer.cancel();
+            if (!me->_gate.is_closed() && has_lag_metric) {
+                me->_lag_metrics_timer.arm(me->_lag_collection_interval());
+            }
+        };
+
+        _lag_metrics_timer.set_callback([this, set_lag_timer]() {
+            ssx::spawn_with_gate(_gate, [this, set_lag_timer] {
+                return collect_consumer_lag_metrics().then(
+                  [this, set_lag_timer] { set_lag_timer(this); });
```

Comment:
it looks like you'd probably want to set `set_lag_timer` even if `collect_consumer_lag_metrics` experiences a problem?